### PR TITLE
After repeated delivery failures, give up on inboxes

### PR DESCRIPTION
**Pull request by Gargron**
**Note:** This can be probably written in a much better way, ideas on how to structure the code are welcome

What makes us give up on an inbox:

- 7 days of successive DeliveryWorker *exhaustions* (i.e. after all natural Sidekiq retries for a job have been exhausted)

What returns an inbox back to normal:

- A successful delivery cancels it out
- An incoming delivery from account of the inbox cancels it out

### CODE REVIEW

File: app/lib/request.rb [hunk 1]
```diff
@@ -85,6 +85,6 @@ def timeout
   end
 
   def http_client
-    HTTP.timeout(:per_operation, timeout).follow
+    HTTP.timeout(:per_operation, timeout).follow(max_hops: 2)
```
Comment (unarist) [inline app/lib/request.rb]: Is this change related to this topic?

**Review by beatrix-bitrot — APPROVED** [review]
nice!

**Comment by beatrix-bitrot** [comment]
not a blocking thing but this would be nice to have as an environment variable and easy too
